frame/support/src/inherent.rs

```diff
@@ -198,15 +270,20 @@ mod tests {
 		const INHERENT_IDENTIFIER: sp_inherents::InherentIdentifier = *b"test1235";
 
 		fn create_inherent(_: &InherentData) -> Option<Self::Call> {
-			Some(CallTest::Something)
+			Some(CallTest::OptionalInherent(true))
 		}
 
 		fn check_inherent(call: &Self::Call, _: &InherentData) -> Result<(), Self::Error> {
 			match call {
-				CallTest::Something => Ok(()),
-				CallTest::SomethingElse => Err(().into()),
+				CallTest::OptionalInherent(true) => Ok(()),
+				CallTest::OptionalInherent(false) => Err(().into()),
+				_ => Ok(())
```

Comment:
why should this accept `CallTest:: NotInherent `?

Reply:
`NotInherent` should never be checked in `check_inherent` so the implementation doesn't matter, but I replaced with unreachable for better testing and readability.